cppcache/src/TcrMessage.cpp

```diff
@@ -838,6 +840,22 @@ void TcrMessage::processChunk(const uint8_t* bytes, int32_t len,
       }
       break;
     }
+    case PUT_DATA_ERROR: {
+      chunkSecurityHeader(1, bytes, len, isLastChunkAndisSecurityHeader);
+      if (nullptr != bytes) {
+        auto input =
+            m_tcdm->getConnectionManager().getCacheImpl()->createDataInput(
+                bytes, len);
+        auto errorString = readStringPart(input);
+
+        if (!errorString.empty()) {
+          setThreadLocalExceptionMessage(errorString.c_str());
+        }
+
+        _GEODE_SAFE_DELETE_ARRAY(bytes);
+      }
+      break;
+    }
     case TcrMessage::GET_ALL_DATA_ERROR: {
```

Comment:
These errors claim to be handled at a higher level, is the handling different than `PUT_DATA_ERROR`?

Reply:
GET_ALL_DATA_ERROR is not a catchall for data errors, it is a specific error response to a get-all data request.

Reply:
My comment had to do with the claim that `GET_ALL_DATA_ERROR`s are handled at a higher level (line 864) and how does this compare to the route taken for `PUT_DATA_ERROR`? i.e. which is correct as they seem to be handled very differently...

Reply:
If I understand you correctly, you're saying PUT_DATA_ERROR handler should eat these exceptions just like the other handlers. However, it seems like we should bubbling these exceptions up to the user. It looks like these aren't even logged in the client log, which seems to make it even more important to pass these back to the user.